### ISSUE TITLE
Live audio visualisation

### ISSUE THREAD

**Issue by SossenSystems** [mozartec/capacitor-microphone#24]
Firstly I would like to say, really cool project, **thanks for your fantastic work!**

Is there a way to capture the data live while the microphone is recording without having to stop the recording first? I am interested in this because I would like to do some kind of audio visualisation, for example like in a WhatsApp voice message.

Thank you very much in advance for your answer!

**Comment by Mozart-Alkhateeb** [comment]
Thanks, I tested WhatsApp on iOS and the Only thing I saw was a timer that starts when you start recording. can you give me info?

**Comment by SossenSystems** [comment]
> Thanks, I tested WhatsApp on iOS and the Only thing I saw was a timer that starts when you start recording. can you give me info?

Yes, i also want to implement a timer, but it isn't a real problem because they are a lot of workarounds.

The biggest issue for me is like that:![image](https://github.com/mozartec/capacitor-microphone/assets/33610430/922c7a68-4d4d-435f-a5d7-dee21cae6730)
So i want to create on recording a live sound wave. For me that means I have to be able to access the recording in real time (while recording) via some kind of event listener.

**Comment by SossenSystems** [comment]
Thank you so much! 😊

On the web, for example, there are many frameworks/plugins that make exactly this possible. The way a developer implements this in their Ionic Capacitor app should be left up to them. I think, the live stream from the microphone with data content alone would be enough to be able to build something like this.

Interestingly, someone has already tried something like this with your repo: https://github.com/simonschmalfeld/capacitor-microphone/commit/36bb2f9e9cdc2e532011fd9fe4c450de3f144aff

**Comment by SossenSystems** [comment]
Any new news? @Mozart-Alkhateeb

**Comment by Mozart-Alkhateeb** [comment]
This is a little bit more complicated than I thought at least for android, haven't checked iOS yet.

I use Media Recorder for the underlying native implementation on Android, what you are asking for might require the usage of Audio  Recorder (i.e rewriting most of the code) and you're still far from VoIP.

It is a feature that I might add and will be experimenting with in my freetime. 
If you need this for a project at this time, then you should be looking for a different plugin.
